relayer/crates/starknet-chain-components/src/types/cosmos/client_state.rs

```diff
@@ -101,3 +109,244 @@ impl From<CometClientState> for Any {
         IbcCometClientState::from(client_state).into()
     }
 }
+
+pub struct EncodeLeafOp;
+
+#[cgp_provider(MutEncoderComponent)]
+impl<Encoding, Strategy> MutEncoder<Encoding, Strategy, LeafOp> for EncodeLeafOp
+where
+    Encoding: CanEncodeMut<Strategy, Product![u32, u32, u32, u32, Vec<u8>]>,
+{
+    fn encode_mut(
+        encoding: &Encoding,
+        proof_specs: &LeafOp,
+        buffer: &mut Encoding::EncodeBuffer,
+    ) -> Result<(), Encoding::Error> {
+        let LeafOp {
+            hash,
+            prehash_key,
+            prehash_value,
+            length,
+            prefix,
+        } = serde_json::to_value(proof_specs)
+            .and_then(serde_json::from_value)
+            .unwrap();
+        //.map_err(|_| Encoding::raise_error("invalid connection end"))?;
```

Comment:
do you need to return these errors?

Reply:
Oh good catch will fix those

Reply:
Done in 421fe1fb11ff66b6b39b9353b73f52ee40da5859